pkg/infrastructure/attribute_manager.go

```diff
@@ -47,12 +52,27 @@ func readInfrastructure() (infraattributes.Infrastructure, error) {
 	if err != nil {
 		return attributes, err
 	}
+	if err = validateStructure(content, ValidationPath); err != nil {
+		return attributes, err
+	}
 	if err = json.Unmarshal(content, &attributes); err != nil {
 		return attributes, errors.Wrap(err, "could not parse infrastructure json")
 	}
 	return attributes, nil
 }
 
+func validateStructure(bytes []byte, taxonomySchema string) error {
```

Comment:
Should this be a general helper function provided by the taxonomy, or is it truly specific to the attribute manager?

Reply:
It's just a wrapper for the function in `validate ` package